### PR TITLE
Use StimulusJS for vomit confirmations in Internal

### PR DESCRIPTION
**Pull request by msarit**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Use StimulusJS for vomit confirmations in Internal.
NOTE: due to merging master branch in, I had to resolve some ESLint issues in other files. This is why the #files-changed is more than the files related to this PR

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/pull/7082

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
none

## Added tests?

- [ ] yes
- [X] no, because they aren't needed (at this time, per Jacob)
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [X] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Relieved Denzel

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -931,6 +932,7 @@
     t.datetime "updated_at", null: false
     t.bigint "user_id"
     t.index ["type_of"], name: "index_response_templates_on_type_of"
+    t.index ["user_id", "type_of"], name: "index_response_templates_on_user_id_and_type_of"
```
Comment (Zhao-Andy) [inline db/schema.rb:935]: Same with this?

File: db/schema.rb [hunk 2]
```diff
@@ -307,7 +309,6 @@
     t.string "slug"
     t.string "status", default: "active"
     t.datetime "updated_at", null: false
-    t.index ["slug"], name: "index_chat_channels_on_slug", unique: true
```
Comment (Zhao-Andy) [inline db/schema.rb:310]: I think this shouldn't be in this PR?

File: app/javascript/internal/controllers/reaction_controller.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          if (json.outcome === 'Success') {
+            this.element.remove();
+            document.getElementById(`js__reaction__div__hr__${id}`).remove();
+          } else {
+            alert(json.error);
+          }
+        })
+        .catch((error) => {
+          alert(error);
+        }),
```
Comment (Zhao-Andy) [inline app/javascript/internal/controllers/reaction_controller.js:36]: Nice simple error handling here :)

File: db/schema.rb (hunk 2 above)
Comment (msarit) [reply db/schema.rb:310]: Yeah these somehow found their way in 🤷‍♀️ They weren't my changes. I will fix.

File: app/javascript/internal/controllers/reaction_controller.js (hunk 3 above)
Comment (msarit) [reply app/javascript/internal/controllers/reaction_controller.js:36]: 😸

Comment (msarit) [reply app/javascript/internal/controllers/reaction_controller.js:36]: 😸

**Pull request merged**